### PR TITLE
Fixed some bugs in JS Refactoring

### PR DESCRIPTION
**Pull request by navch**
Fixed some bug in JS Refactoring i.e.

1) Not allowing to rename some keywords
2) If someone tries to convert a named function to arrow function, earlier we used to convert surrounded anonymous function, now if cursor is inside named function then we will show error.
3) Handling default params in case of convert to arrow function

  Please Review @petetnt @swmitra @nethip @boopeshmahendran @saurabh95

### CODE REVIEW

**Comment by nethip** [comment]
@navch Could you resolve this conflicts so that we can go ahead merging the changes.

**Comment by navch** [comment]
Done @nethip

File: src/extensions/default/JavaScriptRefactoring/WrapSelection.js [hunk 1]
```diff
@@ -121,21 +121,52 @@ define(function (require, exports, module) {
...
+        if (!funcExprNode.body) {
+            return;
+        }
+
         var noOfStatements = funcExprNode.body.body.length,
             selectedText = current.text.substr(funcExprNode.start, funcExprNode.end - funcExprNode.start),
-            param = [];
+            param = [],
+            dontChangeParam = false,
+            numberOFParams = funcExprNode.params.length,
```
Comment (nethip) [inline src/extensions/default/JavaScriptRefactoring/WrapSelection.js]: nit: `numberOfParams`

Comment (navch) [reply src/extensions/default/JavaScriptRefactoring/WrapSelection.js]: 👍

**Pull request merged**